### DISCUSSION TITLE
BERT fine tuning from the code

### DISCUSSION THREAD

**Discussion by SoimulPatriei** [explosion/spaCy#8495] [Help: Coding & Implementations] [answered]
Hi,
I have annotated some data with named entities, and I would like to fine-tune the BERT model to the new data, then save the trained model, load the model, and automatically annotate new data. According to the documentation for fine-tuning, we have to issue the following command :

python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./dev.spacy

 However, I would like to perform this step from code, not from outside. How is this possible?
 
 Thanks!
Eduard

**Comment by adrianeboyd** [comment, marked as answer]
Sure, you can see the internals of all the CLI commands in this directory:

https://github.com/explosion/spaCy/tree/master/spacy/cli

You can call the `_cli` functions directly, or copy and adapt the code for your own purposes.

**Reply by SoimulPatriei** [reply]
Thanks!
